packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts

```diff
@@ -1582,32 +1758,30 @@ class ObjectsInRedisClient {
 
         this.checkFileRights(id, oldName, options, CONSTS.ACCESS_WRITE, (err, options, meta) => {
             if (err) {
-                typeof callback === 'function' && callback(err);
+                return tools.maybeCallbackWithError(callback, err);
             } else {
                 if (!options.acl.file.write) {
-                    typeof callback === 'function' && callback(ERRORS.ERROR_PERMISSION);
+                    return tools.maybeCallbackWithError(callback, ERRORS.ERROR_PERMISSION);
                 } else {
                     this._rename(id, oldName, newName, options, callback, meta);
                 }
             }
         });
     }
 
-    renameAsync(id, oldName, newName, options) {
-        return /** @type {Promise<void>} */ (
-            new Promise((resolve, reject) =>
-                this.rename(id, oldName, newName, options, err => (err ? reject(err) : resolve()))
-            )
+    renameAsync(id: string, oldName: string, newName: string, options: CallOptions): Promise<void> {
+        return new Promise<void>((resolve, reject) =>
```

Comment:
This `void` should be inferred from the return type of `renameAync`. No need to specify it twice.